### DISCUSSION TITLE
Discrepancy between X11 window id and i3 id

### DISCUSSION THREAD

**Discussion by ejovo13** [i3/i3#6188] [Q&A]
I had a question about the Window ID that's reported by i3's `get_tree` command because I noticed a discrepancy with what `xwinfo` reports.

In section 6 of i3's user guide: ([i3: i3 User’s Guide](https://i3wm.org/docs/4.2/userguide.html)) the window id section says:

```

id

    Compares the X11 window ID, which you can get via xwininfo for example.

```

However, if we compare the id that I3 reports for my focused window:

```
{ "name": "tmux ~", "id": 93940309207472, "type": "con", "class": "Terminator", "focused": true, "output": "eDP", "sticky": false, "floating": "auto_off", "nodes": [] }
```

with what X11 reports:

```
32653082  # 0x1f23f1a
```

(Confirmation that we are dealing with the same window:
```
xwininfo: Window id: 0x1f23f1a "tmux ~"
```
we see a clear difference.

I imagine that the id that i3 is reporting can be used to retrieve the original X11 id. Would it be possible to point me in the right direction to figure out how to do so? I've tried searching through the docs and even took a peek at the source code but I'm not able to answer this through googling or by myself.

In other words, how exactly is i3's window id computed and how can that be used to retrieve the X11 id?

Cheers,
Ejovo

**Comment by ejovo13** [comment]
So, after closely inspecting the original payload we notice two things:

- `con_id` no longer exists
- There is an undocumented field `window`:

```
{
  "name": "tmux a -t ext ~",
  "id": 93940308944560,
  "type": "con",
  "class": "Terminator",
  "window": 31457287,
  "focused": true,
  "output": "eDP",
  "sticky": false,
  "floating": "auto_off",
  "nodes": []
}
```

After running a `xwininfo` command we corroborate the fact that the `window` field corresponds to the X11 window  id and the `id` field must refer to the `con_id` that appears in the docs.

In other words, the fields that are mentioned in documentation map to the following real payload of i3:

```
con_id => id
id => window
```

##  Confirmation:

```
$ xwininfo -id 31457287
xwininfo: Window id: 0x1e00007 "tmux a -t ext ~"
```

**Reply by orestisfl** [reply]
The documentation for the `GET_TREE` message response is in https://i3wm.org/docs/ipc.html#_tree_reply
